R/user_utils.R

```diff
@@ -87,6 +87,74 @@ convert_blanks_to_na.data.frame <- function(x) { # nolint
   x
 }
 
+
+#' Convert NAs Into Blank Strings
+#'
+#' Turn `NA`s to blank strings .
+#'
+#' @param x Any R object
+#'
+#' @details
+#' The default methods simply returns its input unchanged. The `character` method
+#' turns every instance of `NA_character_` or `NA` into `""` while preserving *all* attributes.
+#' When given a data frame as input the function keeps all non-character columns
+#' as is and applies the just described logic to `character`
+#' all attributes such as labels are preserved.
+#'
+#' @return An object of the same class as the input
+#'
+#' @author Sadchla Mascary
+#'
+#' @family utils_fmt
+#' @keywords utils_fmt
+#'
+#' @export
+#'
+#' @examples
+#' library(tibble)
+#'
+#' convert_na_to_blanks(c("a", "b", NA, "d", NA))
+#'
+#' df <- tibble(
+#'   a = structure(c("a", "b", NA, "c"), label = "A"),
+#'   b = structure(c(1, NA, 21, 9), label = "B"),
+#'   c = structure(c(TRUE, FALSE, TRUE, TRUE), label = "C"),
+#'   d = structure(c(NA, NA, "s", "q"), label = "D")
+#' )
+#' print(df)
+#' convert_na_to_blanks(df)
+convert_na_to_blanks <- function(x) {
```

Comment:
Could you implement `convert_na_to_blanks()` as S3 generic like `convert_blanks_to na()`?

I.e., the following calls should work.
```
convert_na_to_blanks(list(NA_character_, "1"))
convert_na_to_blanks(tibble(AVALC = c("1", NA_character_)))
```

For this you need to update `convert_na_to_blanks()` such that `UseMethod("convert_na_to_blanks")` is called and rename the other functions such that the form `<method name>.<class name>()` is used, e.g., `convert_na_to_blanks.default()` or `convert_na_to_blanks.data.frame()`.

See https://adv-r.hadley.nz/s3.html#s3-methods for details on S3 generics.